### PR TITLE
MC truth classes renaming

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Historically some classes used at SIM step have names started from "G4". It is incorrect, because it is the namespace of the Geant4 toolkit. In this PR several classes are renamed in order to have better correspondence between perpose of these clases used for MC truth handling within SIM step. Regression is expected, should be no change of any result.

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
FYI, @fabiocos , @bsunanda

**Comment by civanch** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by civanch** [comment]
please test

File: SimG4Core/Notification/src/MCTruthUtil.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                                motherInfo->getIDLastVolume(),
+                                motherInfo->caloSurfaceParticlePID(),
+                                motherInfo->caloSurfaceParticleP());
+  }
+
+  // for Run1 and Run2
+  if (motherInfo->hasCastorHit()) {
+    trkInfo->setCastorHitPID(motherInfo->getCastorHitPID());
+  }
+
```
Comment (fabiocos) [inline SimG4Core/Notification/src/MCTruthUtil.cc:56]: @civanch as far as I can see, this is basically a renaming of ```NewTrackAction```. My recent addition for BTL should be plugged here

**Comment by civanch** [comment]
This PR is closed and substituted by #41370

**Pull request closed without merging**